src/utils/svg.ts

```diff
@@ -124,3 +150,96 @@ export function initFonts(svg: Element) {
   }
   return false
 }
+
+/**
+ * Retrieves the attribute of an SVG and appends a suffix to ensure uniqueness.
+ * 
+ * @param attr String representing the name of the attribute to modify
+ * @param desc SVG to perform the modification on
+ * @param count Number representing the current value of the counter
+ */
+ export function replaceAttr(attr: string, desc: Element, count: number) {
+  var iri = desc.getAttribute(attr);
+  if (/^\s*#/.test(iri!)) { // Check if iri is non-null and internal reference
+    iri = iri?.trim()!;
+    desc.setAttribute(attr, iri! + suffix + count)
+  }
+}
+
+/**
+ * "Mangles" the SVG by appending a SVG-specific suffix to each element's ID to ensure
+ * each SVG component has an ID unique from other SVGs.
+ * 
+ * @param svg SVG element to "mangle"
+ */
+export function mangleSVG(svg: Element) {
+  // Variable declarations
+  let tagLabel = "";
+  let idRegex = /url\("?#([a-zA-Z][\w:.-]*)"?\)/g;
+  let iri_tag_map = new Map<string, number>();
+  let id_map = [];
+  let iriProperties = [];
+  var currentProp;
+  var idElem;
+  let ranVal = Math.random();
+
+  // Retrieve all id's from SVG
+  let allIdElements = svg.querySelectorAll('[id]');
+  let len = allIdElements.length;
+  let iriProp = new Array;
+
+  // Iterate through id's, creates mapping of elems with iri properties (includes mask and others)
+  if (allIdElements.length >= 0) {
+    for (let i = 0; i < len; i++) {
+      tagLabel = allIdElements[i].localName;
+      id_map.push(allIdElements[i].id);
+      if (tagLabel in IRI_TAG_PROPERTIES_MAP) {
+        iri_tag_map.set(tagLabel, 1);
+      }
+    }
+
+    // Properties list creation
+    for (let props of iri_tag_map.keys()) {
+      let lengthProp = iriMap.get(props)?.length!;
+      currentProp = iriMap.get(props);
+      for (let k = 0; k < lengthProp; k++) {
+        if (iriProp.indexOf(currentProp![k]) < 0) {
+          iriProperties.push(currentProp![k]);
+        }
+      }
+    }
+    if (iriProperties.length >= 0) {
+      iriProperties.push('style');
+    }
+
+    //Replacing the actual id's begis here
+    let propertyName = "";
+    let value = "";
+    let newValue = "";
+    let descElem = svg.querySelectorAll('*');
```

Comment:
This is fine, but does hit every element in the SVG. Would be a future investigation to see if there is a way to query only the elements with the relevant attributes so that you don't have to individually check later.